src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -36,6 +37,7 @@ public static partial class Environment
             }
             return results;
         });
+        private static readonly bool IsMac = Interop.Sys.GetUnixName() == "OSX";
```

Comment:
As with Mono, we always return PlatformID.Unix on non-Windows. We could revisit that, but it would be a change from both Mono and what coreclr's PAL had.